src/py4vasp/calculation/_exciton.py

```diff
@@ -0,0 +1,147 @@
+# Copyright © VASP Software GmbH,
+# Licensed under the Apache License 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+import numpy as np
+
+from py4vasp import _config, calculation, exception
+from py4vasp._third_party import view
+from py4vasp._util import documentation, import_, index, select
+from py4vasp.calculation import _base, _structure
+
+pretty = import_.optional("IPython.lib.pretty")
+
+
+_INTERNAL = "1"
+
+class Exciton(_base.Refinery, _structure.Mixin, view.Mixin):
+    """This class accesses exciton charge densities of VASP.
+
+    The exciton charge densities can be calculated via the BSE/TDHF algorithm in
+    VASP. With this class you can extract these charge densities.
+    """
+
+    @_base.data_access
+    def __str__(self):
+        _raise_error_if_no_data(self._raw_data.exciton_charge)
+        grid = self._raw_data.exciton_charge.shape[1:]
+        excitons = self._raw_data.exciton_charge.shape[0]
+        topology = calculation.topology.from_data(self._raw_data.structure.topology)
+        return f"""exciton charge density:
+    structure: {pretty.pretty(topology)}
+    grid: {grid[2]}, {grid[1]}, {grid[0]}
+    excitons : {excitons}"""
+
+    @_base.data_access
+    def to_dict(self):
+        """Read the exciton density into a dictionary.
+
+        Returns
+        -------
+        dict
+            Contains the supercell structure information as well as the exciton
+            charge density represented on a grid in the supercell.
+        """
+        _raise_error_if_no_data(self._raw_data.exciton_charge)
+        result = {"structure": self._structure.read()}
+        result.update(self._read_density())
+        return result
+
+    def _read_density(self):
+        density = self.to_numpy()
+        if self._selection:
```

Comment:
I don't think self._selection is set in your case. In the density, this is used to distinguish between different densities (charge, kinetic energy) and you would set it with `calc.density.plot('charge')` or `calc.density.plot('tau')`. In your case there is only a single possibility to get the exciton density, isn't there?